src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/DependencyTestDriver.cs

```diff
@@ -94,7 +96,7 @@ public async Task UpdatePinnedDependenciesAsync()
         {
             string testVersionDetailsXmlPath = Path.Combine(RootExpectedOutputsPath, VersionFiles.VersionDetailsXml);
             string versionDetailsContents = File.ReadAllText(testVersionDetailsXmlPath);
-            IEnumerable<DependencyDetail> dependencies = _gitFileManager.ParseVersionDetailsXml(versionDetailsContents, false);
+            IEnumerable<DependencyDetail> dependencies = _versionDetailsParser.ParseVersionDetailsXml(versionDetailsContents, false);
```

Comment:
If we got rid of this (and just passed in an XmlDoc instead), could we get rid of the ParseVersionDetailsXml interface method that just takes a string?

Reply:
Actually, it's the other way around. There are quite a few places which just open a file and get a string so it would be cumbersome to make them turn it into an XML. There are just a few that do "XML from URI" directly which require the latter and those would have to be changed.

So this new class needs `string->dependencies` and making the XML step in-between public didn't cost me anything so I just went with this path of least resistance.